WcaOnRails/app/views/persons/_regional_records.html.erb

```diff
@@ -0,0 +1,50 @@
+<% results = @results.select do |result|
+     ([result.regionalSingleRecord, result.regionalAverageRecord] & record_types).any?
+   end %>
+
+<% if results.any? %>
+  <div class="records">
+    <h3 class="text-center"><%= title %></h3>
+    <div class="table-responsive">
+      <table class="table table-striped">
+        <thead>
+          <tr>
+            <th class="event"><%= t 'competitions.results_table.event' %></th>
+            <th class="single"><%= t 'common.single' %></th>
+            <th class="average"><%= t 'common.average' %></th>
+            <th class="competition"><%= t 'persons.show.competition' %></th>
+            <th class="round"><%= t 'competitions.results_table.round' %></th>
+            <th class="solves" colspan="5"><%= t 'common.solves' %></th>
+          </tr>
+        </thead>
+        <tbody>
+          <% results.group_by(&:event).each do |event, results| %>
```

Comment:
Wow, good research! I suspect this means it could change in a future version of ruby. I see 2 options:

1. Leave this as is, even if the behavior changes in the future version of ruby, it's not a catastrophe.
2. We could implement our own `sorted_group_by`, mostly so future readers of the code don't worry about this as I have.

I think 1. is a fine solution here, but do whatever you like!